packages/api/amplify_api/lib/src/graphql/web_socket/blocs/web_socket_bloc.dart

```diff
@@ -369,9 +512,29 @@ class WebSocketBloc with AWSDebuggable, AmplifyLoggerMixin {
     return subBloc;
   }
 
-  // Sends an exception to all sub blocs and cleans up this bloc
-  Future<void> _sendExceptionToBlocs(Exception e) async {
-    _emit(_currentState.failed());
+  /// Takes a WebSocketEvent and sends it to the corresponding sub bloc
+  Stream<WebSocketState> _sendEventToSubBloc(WsSubscriptionEvent event) async* {
+    final id = event.subscriptionId;
+    assert(
+      _currentState.subscriptionBlocs.containsKey(id),
+      'Bloc is missing subscription for $id',
+    );
+    _currentState.subscriptionBlocs[id]!.add(event);
+  }
+
+  /// GET request on the configured AppSync url via the `/poll` endpoint
+  Future<AWSBaseHttpResponse> _sendPollRequest() {
+    final req = AWSHttpRequest.get(
+      _currentState.pollUri,
+    );
+
+    return req.send(client: _pollClient).response;
+  }
+
+  /// Sends an exception to all sub blocs and cleans up this bloc
+  Future<void> _shutdownWithException(Exception e) async {
```

Comment:
I was testing this scenario and noticed that when I hit this shutdown there is no way to establish a new subscription. While I would not expect re-connect to continue once we get here, if user cancels the original subscription, recovers network, tries to create subscription then it never gets established, which I was not expecting.